pkg/roachprod/cloud/gc_aws_keypairs.go

```diff
@@ -316,3 +326,105 @@ func GCAWSKeyPairs(l *logger.Logger, dryrun bool) error {
 	}
 	return nil
 }
+
+// gc garbage-collects expired clusters, unused SSH key pairs in a
+// single AWS account
+func gc(l *logger.Logger, dryrun bool) error {
+	errorsChan := make(chan error, 8)
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		errorsChan <- gCAWSKeyPairs(l, dryrun)
+	}()
+
+	cld, _ := ListCloud(l, vm.ListOptions{IncludeEmptyClusters: true, IncludeProviders: []string{aws2.ProviderName}})
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		errorsChan <- GCClusters(l, cld, dryrun)
+	}()
+
+	go func() {
+		wg.Wait()
+		close(errorsChan)
+	}()
+
+	var combinedError error
+	for err := range errorsChan {
+		combinedError = errors.CombineErrors(combinedError, err)
+	}
+
+	return combinedError
+}
+
+// getTemporaryCredentialsByAssumingRole uses AWS STS to get temporary AWS credentials by assuming an IAM role
+func getTemporaryCredentialsByAssumingRole(
+	roleArn, roleSessionName, region string,
+) (*types.Credentials, error) {
+	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion(region))
+	if err != nil {
+		return nil, errors.Wrap(err, "GetCredentialsByAssumingRole: failed to load config")
+	}
+
+	stsClient := sts.NewFromConfig(cfg)
+	tmpCredentials, err := stsClient.AssumeRole(context.TODO(), &sts.AssumeRoleInput{
+		RoleArn:         aws.String(roleArn),
+		RoleSessionName: aws.String(roleSessionName),
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "GetCredentialsByAssumingRole: failed to assume role")
+	}
+
+	return tmpCredentials.Credentials, nil
+}
+
+// setTemporaryCredentialsByAssumingRole fetches temporary aws credentials by assuming an IAM role and sets
+// amazon.AWSAccessKeyParam, amazon.AWSSecretParam, amazon.AWSTempTokenParam in environment variables
+func setTemporaryCredentialsByAssumingRole(roleArn, roleSessionName, region string) error {
+	tmpCredentials, err := getTemporaryCredentialsByAssumingRole(roleArn, roleSessionName, region)
+	if err != nil {
+		return err
+	}
+	_ = os.Setenv(amazon.AWSAccessKeyParam, *tmpCredentials.AccessKeyId)
+	_ = os.Setenv(amazon.AWSSecretParam, *tmpCredentials.SecretAccessKey)
+	_ = os.Setenv(amazon.AWSTempTokenParam, *tmpCredentials.SessionToken)
+	return nil
+}
+
+// GCAWS garbage collects expired clusters, unused SSH keys for a single or multiple AWS accounts
+func GCAWS(l *logger.Logger, dryrun bool) error {
+	provider := vm.GetProviderByName(aws2.ProviderName)
+	var awsAccountIds []string
+	if awsProviderInstance, ok := provider.(*aws2.Provider); ok {
+		awsAccountIds = awsProviderInstance.AccountIds
+	}
+
+	// if accountIds are not provided performs cleanup on a single AWS account configured in ${HOME}/.aws/credentials file
+	if len(awsAccountIds) == 0 {
+		return gc(l, dryrun)
+	}
+
+	// performs garbage collection on all aws account one by one
+	var combinedErrors error
+	for _, accountId := range awsAccountIds {
+		roleArn := fmt.Sprintf("arn:aws:iam::%s:role/shailendra.patel-test-role", accountId)
```

Comment:
Temporary role I assume? We should make a more proper one before merging.

Reply:
Created a new role `arn:aws:iam::<account-id>:role/roachprod-gc-cronjob`